src/ir/analysis/derive.rs

```diff
@@ -156,15 +159,26 @@ impl<'ctx> CannotDerive<'ctx> {
 
         trace!("ty: {:?}", ty);
         if item.is_opaque(self.ctx, &()) {
-            if !self.derive_trait.can_derive_union() &&
-                ty.is_union() &&
+            if ty.is_union() &&
                 self.ctx.options().rust_features().untagged_union
             {
```

Comment:
Maybe:

```rust
let can_derive = self.derive_trait.can_derive_union();
if can_derive != CanDerive::Yes {
    trace!(..);
    return can_derive;
}
```

Reply:
I agree it's a little cleaner. But, our trace messages are less precise this way. Is there a good way to report the trace message differently if `can_derive` is `Manually` vs `No`?

Reply:
include `can_derive` in the trace message?

Reply:
I guess it's not a fancy customized message, but that's easy enough :)